src/arithmetic/big_native/primes.rs

```diff
@@ -0,0 +1,518 @@
+//! This module entirely and shamelessly copypasted from [num-bigint-dig] crate with slight
```

Comment:
isn't it possible to call this library API instead of copying it? 
on another note: prime numbers generation is non-trivial and doing it wrong can jeopardise an entire cryptosystem. copying such code means that if upstream finds a bug we will not know about it. I am also cannot be sure that the primes generation used here is secure. I think that for now - we should clarify (readme? ) that moving to native Rust bigint should be considered experimental for this reason , what do you think ?

Reply:
Unfortunately, it's not possible to reuse this library without copying it. I will put warning in readme.